core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2080,14 +2090,28 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
       maxConcurrent: Int
   )(f: O => F2[O2])(implicit F: Concurrent[F2]): Stream[F2, O2] = {
 
-    val init = ().pure[F2]
+    val init = F.unit
 
     def send(ch: Channel[F2, F2[Either[Throwable, O2]]], release: F2[Unit]) =
       (el: Either[Throwable, O2]) => release <* ch.send(el.pure[F2])
 
     parEvalMapAction(maxConcurrent, f)((ch, release) => init.as(send(ch, release)))
```

Comment:
Could we just reduce this to `F.pure(send(ch, release))` and drop the `init` val?

Reply:
Same comment as above, maybe it's a stack-safety issue?

Reply:
It wasn't intended to add stack-safety. Instead, it's written this way to resemble the logic in parEvalMap

Reply:
So yeah, if it raises question of why it is written this way, I have no argument against `F.pure(send(ch, release))`

Reply:
Thanks @nikiforo, much appreciated!

Reply:
Yeah, I am also quite curious about that approach. Does not seem to be stack-safer than just `F.pure(...)`.

Reply:
It was probably a stupid thing of me to say in this case :) but tricks with `F.unit` for stack-safety are not uncommon in FS2.